ARM: dts: Update Bluefin device trees

HatsyRei opened the pull request:
WI: [AB#2959883](https://dev.azure.com/ni/DevCentral/_workitems/edit/2959883/)

Changes required to build ni-bluefin.dts with updated [zynq-7000.dtsi](https://github.com/ni/linux/blob/nilrt/master/6.6/arch/arm/boot/dts/xilinx/zynq-7000.dtsi) device tree bindings.

Testing: Manually built kernel with changes and decompiled resulting ni-bluefin.dtb via `dtc` to ensure changes are applied.

gratian commented on arch/arm/boot/dts/xilinx/ni-zynq.dtsi:
```diff
@@ -124,108 +105,6 @@
-			interrupts = <0 51 4>;
-		};
-
-		fpgaperipheral@f8007000 {
```
I don't see 'fpgaperipheral' listed in the parent zynq-7000.dtsi. I guess the base address matches 'devcfg@f8007000' so removing it here may be the right thing but I can't tell since it's not listed in the commit message.

HatsyRei replied:
The devcfg node does appear to represent the Xilinx FPGA manager ([ref](https://www.kernel.org/doc/Documentation/devicetree/bindings/fpga/xilinx-zynq-fpga-mgr.txt)). Will amend the commit message to clarify this.

gratian commented on arch/arm/boot/dts/xilinx/ni-zynq.dtsi:
```diff
@@ -235,38 +114,41 @@
+};
 
-		usb0: usb@e0002000 {
-			usb-phy = <&usb_phy0>;
```
I don't think the usb-phy attribute comes from zynq-7000.dtsi since the usb_phy0 node is declared in this file, see lines 14-17. I suspect you need to keep this.

gratian commented on arch/arm/boot/dts/xilinx/ni-zynq.dtsi:
```diff
@@ -235,38 +114,41 @@
+	status = "okay";
 
-		usb1: usb@e0003000 {
-			usb-phy = <&usb_phy1>;
```
Same comments as above. In general, make sure all the stuff you are removing actually has an equivalent in the common zynq-7000.dtsi we are including and is not just being extended here.

HatsyRei commented on arch/arm/boot/dts/xilinx/ni-zynq.dtsi:
```diff
@@ -235,38 +114,41 @@
+	};
+};
+
+&usb0 {
+	usb-phy = <&usb_phy0>;
+};
 
+&usb1 {
+	usb-phy = <&usb_phy1>;
 };
```
@gratian Regarding your comments on the usb-phy attributes, I modified them to be added to the usb nodes by referencing the usb phandles. Should we keep the original bindings?

gratian replied:
OK. I think that's fine but it wasn't immediately obvious why the got removed.

That tells me you should probably split this into its own commit with a commit message that explains why they are not needed anymore (i.e. they're "added to the usb nodes by referencing the usb phandles ...")

HatsyRei replied:
Split these (along with the same change for i2c0) into a separate commit.

gratian commented:
Also, can you rebase this PR on the latest nilrt/master/6.6. Sorry, I've pulled another PR before yours (kernel upgrade).

HatsyRei committed:
ARM: dts: ni-zynq: Drop obsolete bindings

Drop device tree bindings which are already declared in zynq-7000.dtsi.
This includes CPU, CAN, FPGA, PL353 SMC, NAND, GPIO, I2C and SPI
bindings.

'fpgaperipheral' node, now 'devcfg' in zynq-7000.dtsi describes the
Xilinx Zynq FPGA Manager.

Signed-off-by: Hatsy Rei <yan.huan.chng@emerson.com>

HatsyRei committed:
ARM: dts: ni-zynq: Enable zynq-7000 SMC, NAND nodes

Enable PL353 SMC and NAND device tree nodes declared in zynq-7000.dtsi.
Indicate via 'nand-on-flash-bbt' property that bad block table is stored
on the nand flash, and 'nand-ecc-mode' property that on-die ECC is used.

Signed-off-by: Hatsy Rei <yan.huan.chng@emerson.com>

HatsyRei committed:
ARM: dts: ni-zynq: Extend I2C, USB nodes via phandle

Avoid redeclaring these nodes and just extend them via phandles.

Signed-off-by: Hatsy Rei <yan.huan.chng@emerson.com>

HatsyRei committed:
ARM: dts: ni-bluefin: Update AMBA and GPIO bindings

zynq-7000.dtsi contains updated AMBA and GPIO device tree node phandles.
Adopt these changes by referencing i2c0 phandle instead of amba node for
overriding i2c bindings, and refer to gpio0 phandle instead of gpio.

Signed-off-by: Hatsy Rei <yan.huan.chng@emerson.com>
